doc/specs/stdlib_linalg.md

```diff
@@ -206,3 +206,57 @@ program demo_outer_product
     !A = reshape([3., 6., 9., 4., 8., 12.], [3,2])
 end program demo_outer_product
 ```
+
+## `empty` - `empty` creates a new vector or matrix of `integer` type and given shape, without initializing values.
+
+### Status
+
+Experimental
+
+### Class
+
+Pure function.
+
+### Description
+
+`empty` creates a new vector or matrix of `integer` type and given shape, without initializing values.
+
+`empty`, unlike `zeros`, does not set the array values to zero, and may therefore be marginally faster. On the other hand, it requires the user to manually set all the values in the array, and should be used with caution.
+
+### Syntax
+
+For vector:
+`result = [[stdlib_linalg(module):empty(interface)]](dim)`
+
+For matrix:
+`result = [[stdlib_linalg(module):empty(interface)]](dim1, dim2)`
+
+### Arguments
+
+`dim/dim1`: Shall be an `integer` type.
+
+`dim2`: Shall be an `integer` type.
+
+### Return value
+
+Returns a new vector or matrix of `integer` type and given shape, without initializing values.
+
+### Example
+
+```fortran
+program demo_linlag_empty
+    
+    use stdlib_linlag, only: empty
+    implicit none
+    integer, allocatable :: i(:,:)
+
+    print *, empty(2, 1)
+    print *, 0.0*empty(2)                               !! 0.00000000       0.00000000
+    print *, empty(2) + empty(2)
+    print *, (0.1, 0.1)*empty(2) + (0.2, 0.2)*empty(2)
+
+    i = empty(2,2)
```

Comment:
Thanks, this is an interesting question. I think that for `empty`, it mainly serves `stdlib_linalg`, with a clearer purpose and simpler use.
It has the following advantages:
1. Can quickly create vectors and matrices;
2. You can `allocate` and repeat to `allocate` a certain `allocatable` `vector/matrix`.
3. It can directly participate in the calculation **as a function**, which has no application requirements in most cases.

The disadvantages of `allocate` are:
1. It is not a function.
2. Allocation cannot be repeated. (This may be an advantage in terms of rigor)
example:
```fortran
real, allocatable :: A(:, :)
A = empty(2,2)
A = empty(3,3) !! It is OK.
---
real, allocatable :: A(:, :)
allocate(A(2,2))
allocate(A(3,3)) !! It is an error: Attempting to allocate already allocated variable'A'.
````

In fact, `empty` is similar to `allocate`, with subtle differences. `empty` is only suitable for the creation of 1, 2 dimensional arrays, which meets the general needs of `stdlib_linalg`, and it can repeat `allocate` for `vector/matrix`, and the readability is different from `allocate`.
_In fact, to a certain extent, I am not clear about the practical applications of `empty`._
It is theoretically faster than creating `zeros` and `ones` (see #478) because there is no assignment process.

Reply:
>  It is theoretically faster than creating zeros and ones (see #478) because there is no assignment process.

This is actually not correct, in the current implementation `zeros`, `ones` and `empty` are identical, as they all require to copy the automatic array to the newly allocated LHS array.